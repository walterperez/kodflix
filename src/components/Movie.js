import React from "react";

export default function Movie (props) {
    return (
        <div className="flex-element">
        <img src={props.cover} alt={`${props.name} Cover`} />
        <div className="box">
          <h1>{props.name}</h1>
        </div>
      </div>
    )
}