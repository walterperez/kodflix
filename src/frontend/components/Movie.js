import React from "react";
import { Link } from "react-router-dom";

export default function Movie(props) {
  return (
    <div className="flex-element">
      <Link to={props.id}>
        <img src={require(`./../common/img/${props.id}.jpg`)} alt={`${props.name} Cover`} />
      </Link>
      <div className="box">
        <h1>{props.name}</h1>
      </div>
    </div>
  );
}
