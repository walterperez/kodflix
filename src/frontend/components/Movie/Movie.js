import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

function Movie(props) {
  return props.id && props.name ? (
    <Link to={props.id} className="flex-element" data-test="Movie">
      <img src={`/movies/covers/${props.id}.jpg`} alt={`${props.name} Cover`} />
      <div className="box">
        <h1>{props.name}</h1>
      </div>
    </Link>
  ) : null;
}

Movie.propTypes = {
  id: PropTypes.string,
  name: PropTypes.string
};

export default Movie;
