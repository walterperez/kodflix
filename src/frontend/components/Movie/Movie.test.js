import React from 'react';
import { MemoryRouter as Router } from 'react-router-dom';
import { configure, shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import checkPropTypes from 'check-prop-types';
import Movie from './Movie';

configure({ adapter: new Adapter() });

describe('Movie Component', () => {
  describe('Props Types', () => {
    it('Must not give warning', () => {
      const expectedProps = {
        id: 'Batman',
        name: 'Batman'
      };
      const errorProps = checkPropTypes(
        Movie.propTypes,
        expectedProps,
        'props',
        Movie.name
      );
      expect(errorProps).toBeUndefined();
    });
  });

  describe('No Props', () => {
    it('No Props', () => {
      const wrapper = shallow(<Movie />);
      expect(wrapper.find(`[data-test='Movie']`).length).toBe(0);
    });

    it('No Props Snapshot', () => {
      const output = renderer.create(<Movie />);
      expect(output).toMatchSnapshot();
    });
  });

  describe('With Props', () => {
    const propsToPass = {
      id: 'Batman',
      name: 'Batman'
    };

    it('expected props', () => {
      const wrapper = shallow(
        <Movie id={propsToPass.id} name={propsToPass.name} />
      );
      expect(wrapper.find(`[data-test='Movie']`).length).toBe(1);
    });

    it('Expected props Snapshot', () => {
      const output = renderer.create(
        <Router>
          <Movie id={propsToPass.id} name={propsToPass.name} />
        </Router>
      );
      expect(output).toMatchSnapshot();
    });
  });
});

export default Movie;
