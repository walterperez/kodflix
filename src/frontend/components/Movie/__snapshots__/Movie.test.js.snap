// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie Component No Props No Props Snapshot 1`] = `null`;

exports[`Movie Component With Props Expected props Snapshot 1`] = `
<a
  className="flex-element"
  data-test="Movie"
  href="/Batman"
  onClick={[Function]}
>
  <img
    alt="Batman Cover"
    src="/movies/covers/Batman.jpg"
  />
  <div
    className="box"
  >
    <h1>
      Batman
    </h1>
  </div>
</a>
`;
